--- /home/bertw/proj/mcu/tronferno-fhem/svn_fhem_trunk/FHEM/00_SIGNALduino.pm	2017-11-23 13:18:48.743293265 +0100
+++ FHEM/00_SIGNALduino.pm	2017-11-26 07:49:53.179299076 +0100
@@ -138,6 +138,7 @@
 			        	."Siro:"
 						."FHT:"
 						."FS20:"
+						."Fernotron:"
 			      		."SIGNALduino_un:"
 					; 
 
@@ -166,6 +167,7 @@
      "22:Siro"					=> '^P72#[A-Fa-f0-9]+',
      "23:FHT"       => "^81..(04|09|0d)..(0909a001|83098301|c409c401)..",
      "24:FS20"      => "^81..(04|0c)..0101a001", 
+     "25:Fernotron"  => "^P77#[A-Fa-f0-9]{3}",  # only the fist of 12 10bit words (module processes the raw data)
      "X:SIGNALduino_un"			=> '^[u]\d+#.*',
 );
 
@@ -1491,13 +1493,25 @@
 			length_min		=> 58,
 			length_max		=> 58,
 		},
-
+    "77" => ## Fernotron shutters and light switches   
+			#MU;P0=-32001;P1=435;P2=-379;P4=-3201;P5=831;P6=-778;D=01212121212121214525252525252521652161452525252525252161652141652521652521652521614165252165252165216521416521616165216525216141652161616521652165214165252161616521652161416525216161652161652141616525252165252521614161652525216525216521452165252525252525;CP=1;O;
+			# the messages received are usual missing 1.5 10 bits words at the end. So the checksum byte is not received.
+ 			# Fernotron protocol is unidirectional. Here we can only receive messages from controllers send to receivers. 
+		{
+			name         => 'Fernotron',
+			id           => '77',   # protocol number
+			one          => [1,-2], # on=400us, off=800us
+			zero         => [2,-1], # of=800us, off=400us
+			start        => [1,-8], # (on=400us, off=3200us. each 10bit word has [1,-8] in front) FIXME: is 'start' the wrong option for this?
+			clockabs     => 400,    # 400us
+			preamble     => 'P77#', # prepend our protocol number to converted message
+			clientmodule => 'Fernotron', # 
+			length_min   => '10', # real minimal message is 120 bit (12 x 10bit words to decode 6 bytes data) 
+			length_max   => '10', # real maximal message is 3360 bit (336 x 10bit words to decode 168 bytes data)
+	     	},
 );
 
 
-
-
-
 sub
 SIGNALduino_Initialize($)
 {
