--- /opt/fhem/FHEM/00_SIGNALduino.pm.orig	2018-09-15 22:15:33.408212652 +0200
+++ /opt/fhem/FHEM/00_SIGNALduino.pm	2018-09-15 22:24:55.819181547 +0200
@@ -140,6 +140,7 @@
 			        	."Siro:"
 						."FHT:"
 						."FS20:"
+						."Fernotron:"
 			      		."SIGNALduino_un:"
 					; 
 
@@ -168,6 +169,7 @@
      "22:Siro"					=> '^P72#[A-Fa-f0-9]+',
      "23:FHT"       => "^81..(04|09|0d)..(0909a001|83098301|c409c401)..",
      "24:FS20"      => "^81..(04|0c)..0101a001", 
+     "25:Fernotron"  => '^P82#.*',
      "X:SIGNALduino_un"			=> '^[u]\d+#.*',
 );
 
@@ -1479,13 +1481,25 @@
 			length_min		=> 58,
 			length_max		=> 58,
 		},
-
+ 
+ 	"82" => ## Fernotron shutters and light switches   
+ 			#MU;P0=-32001;P1=435;P2=-379;P4=-3201;P5=831;P6=-778;D=01212121212121214525252525252521652161452525252525252161652141652521652521652521614165252165252165216521416521616165216525216141652161616521652165214165252161616521652161416525216161652161652141616525252165252521614161652525216525216521452165252525252525;CP=1;O;
+ 			# the messages received are usual missing 12 bits at the end for some reason. So the checksum byte is missing.
+  			# Fernotron protocol is unidirectional. Here we can only receive messages  sent from controllers to shutters.
+ 		{
+ 			name           => 'Fernotron',
+ 			id             => '82',       # protocol number
+ 			one            => [1,-2],     # on=400us, off=800us
+ 			zero           => [2,-1],     # on=800us, off=400us
+ 			float          => [1,-8],     # on=400us, off=3200us. each 10bit word has one [1,-8] in front
+ 			clockabs       => 400,        # 400us
+ 			preamble       => 'P82#',     # prepend our protocol number to converted message
+ 			clientmodule   => 'Fernotron',
+ 			length_min     => '100',     # actual 120 bit (12 x 10bit words to decode 6 bytes data), but last 20 bit are for checksum and truncated by SIGNALduino firmware
+ 			length_max     => '3360',    # 3360 bit (336 x 10bit words to decode 168 bytes data) for full timer message
+ 	     	},
 );
 
-
-
-
-
 sub
 SIGNALduino_Initialize($)
 {
